datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -986,6 +986,26 @@ macro_rules! general_repeat_list {
     }};
 }
 
+/// Array_empty SQL function
+pub fn array_empty(args: &[ArrayRef]) -> Result<ArrayRef> {
+    if args[0].as_any().downcast_ref::<NullArray>().is_some() {
```

Comment:
I think this doesn't catch all nulls. This only works within the test setup because `array_empty(NULL)` creates a NULL array. I think this will fail w/:

```sql
SELECT array_empty(a)
FROM VALUES(NULL, make_array(), make_array(1)) sub (a);
```

I suggest you remove this check and change the else part within `array.iter().map(...)` to `None` (from `Some(true)`).

Reply:
The whole handling of nulls in the array function logic is "maturing" I would say (e.g. https://github.com/apache/arrow-datafusion/issues/7142 and others listed on https://github.com/apache/arrow-datafusion/issues/6980)

Reply:
I agree, but that doesn't mean we have to add more immature code, esp. when the fix is rather simple (I would argue the fixed code is even simpler than the current version).

Reply:
> I suggest you remove this check and change the else part within `array.iter().map(...)` to `None` (from `Some(true)`).

It cannot handle all NULL array cases in this way, but I think we can wait #7142 to make this pr better.